chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+
+[Fevm]
+# This is set to false by default which disables the ETH RPC API. 
+# Please ensure that you set it to true before starting your node.
+  EnableEthRPC = true
+
+[Events]
+# This is set to false by default which disables the Actor Events API. 
+# Please ensure that you set it to true before starting your node.
+  EnableActorEventsAPI = true
+```
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+#### Configuration
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer periodically runs GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**this is the default**) completely disables GC.
+- Any non-zero value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*for example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to (number of Filecoin epochs in a day *2) = 5760).
+
+---
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the `ChainIndexer` config options explained above:**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+These options are now deprecated and will not have any effect if used in your configuration file. Please use the `ChainIndexer` config options as described above.
+
+---
+
+## Migration Guide
+
+Migrating to the new `ChainIndexer` involves several steps to ensure a smooth transition. Here's a guide to help you through the process:
+
+1. **Backup Existing Index Databases**
+   - Before restarting your Lotus node, create a backup of your existing index databases.
+   - These files are located in the `{$LOTUS_PATH/sqlite}` directory.
+   - While not used in the migration process, these backups are crucial for potential rollbacks.
+
+2. **Remove Old Index Files**
+   - After creating backups, remove the SQLite database files for `MsgIndex`, `EthTxIndex`, and `EventIndex` from the `{$LOTUS_PATH/sqlite}` directory.
+
+3. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the `ChainIndexer Config` section above.
+
+4. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately.
```

Comment:
What about the migration time when moving to the ChainIndexer? Do we have some estimates here

Reply:
Or to be clearer, it now sound a little bit like:

- Create a backup
- Remove Old Index
- Update Config
- Restart Lotus Node

And you should be all good. But for both snapshot synced / archival node operators it will take quite some time to do the backfilling?

Reply:
@rjan90 Noted. Will add the timings when I have them.

Reply:
I'm marking this as a TODO in https://github.com/filecoin-project/lotus/pull/12600.  I'm resolving it her so don't have to keep opening up this big PR to fine unresolved conversations.